File: C/50.Consecutive_prime_sum/ValentinDSJ.c
//Euler problem 50
//Consecutive prime sum

#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <stdbool.h>

#define LIMIT 1000000

static bool is_prime(int x)
{
    int y = 5;

    if (x <= 3)
        return true;
    else if ((x % 2 == 0) || (x % 3 == 0))
        return false;
    while ((y * y) <= x) {
        if ((x % y == 0) || (x % (y + 2) == 0))
            return false;
        y += 6;
    }
    return true;
}

static void prime_tab(void)
{
    int *prime_tab = calloc(LIMIT / 4, sizeof(int));
    int length = 0;

    for (int x = 2; x < LIMIT; x++)
        if (is_prime(x)) {
            prime_tab[length] = x;
            length++;
        }
    prime_tab[length] = -1;
    
    int *sum_tab = calloc(length + 1, sizeof(int));

    sum_tab[0] = 0;
    length = 0;
    for (int x = 0; sum_tab[x] < LIMIT; x++) {
        sum_tab[x + 1] = sum_tab[x] + prime_tab[x];
        length++;
    }
    
    int res = 0;
    int temp_res = 0;
    int diff = 0;
    int temp_diff = 0;
    for (int x = length; x > 0; x--) {
        for (int y = 0; y < length; y++) {
            temp_res = sum_tab[x] - sum_tab[y];
            temp_diff = x - y;
            if (temp_res > LIMIT)
                break;
            if (is_prime(temp_res) && (temp_diff > diff)) {
                res = temp_res;
                diff = temp_diff;
            }
        }
    }
    printf("The biggest sum of consecutive primes is: %d\nWith: %d consecutive primes.\n", res, diff);
}

int main(void)
{
    prime_tab();
    return 0;
}

File: Go Lang/050.Consecutive_prime_sum/ValentinDSJ.go
//Euler 50.Consecutive_prime_sum in Go
package  main

import "fmt"

const limit int = 1000000

func is_prime(x int) (res bool) {
	y := 5

	if x <= 3 {
		return true
	} else if (x % 2) == 0 || (x % 3) == 0 {
		return false
	}
	for (y * y) <= x {
		if (x % y) == 0 || (x % (y + 2)) == 0 {
			return false
		}
		y += 6;
	}
	return true
}

func solver() (res int, diff int) {
	prime_tab := []int{}

	for x := 2; x < limit; x++ {
		if (is_prime(x)) {
			prime_tab = append(prime_tab, x)
		}
	}
	
	sum_tab := []int{0}

	for x := 0; sum_tab[x] < limit; x++ {
		sum_tab = append(sum_tab, sum_tab[x] + prime_tab[x])
	}
	
	res = 0
	diff = 0
	temp_res := 0
	temp_diff := 0
	for x := len(sum_tab) - 1; x > 0; x-- {
		for y := 0; y < len(sum_tab); y++ {
			temp_res = sum_tab[x] - sum_tab[y]
			temp_diff = x - y
			if temp_res > limit {
				break;
			}
			if is_prime(temp_res) && temp_diff > diff {
				res = temp_res
				diff = temp_diff
			}
		}
	} 

	return
}

func main() {
	x, y := solver()
	fmt.Printf("The biggest sum of consecutive primes is %d\nWith %d consecutive primes.", x, y)
}
